src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -3,203 +3,218 @@ namespace NServiceBus
     using System;
     using System.Collections.Concurrent;
     using System.Collections.Generic;
-    using System.Runtime.CompilerServices;
     using System.Threading.Tasks;
     using Extensibility;
-    using Persistence;
     using Sagas;
-
+    using Persistence;
+    
     class InMemorySagaPersister : ISagaPersister
     {
+        static readonly CorrelationId NoCorrelationId = new CorrelationId(typeof(object), "", new object());
+        readonly ConcurrentDictionary<Guid, Entry> sagas = new ConcurrentDictionary<Guid, Entry>();
```

Comment:
I don't agree with. Using an extension method is more intention revealing, does not allocate and makes a potential reader of the `InMemorySagaPersister` deal only with the concurrent dictionaries rather than potentially abusive usage of this API.